### ISSUE TITLE
A* Algorithm

### ISSUE THREAD

**Issue by DPigeon** [DPigeon/Heuristic-Indonesian-Puzzle#30]
As a user, I want to create an A* algorithm so that it find the optimal solution path.

As a user, I want to build an A* algorithm from inputs given to give us the proper searched nodes and solution path. The algorithm must follow f(n) = g(n) + h(n) where

- f(n): estimate of total cost along path through n
- g(n): actual cost of path from start to node n 
- h(n): estimate of cost to reach goal from n (heuristic #32)

In summary, it will have to keep track of the depth (g) and the estimate (h). The smaller f(n) will be chosen each iteration. The heuristic h will have to be ran on all nodes to evaluate their estimate.

This is dependent of:
#32 
#34 

Criteria of Acceptance:
- Is the algorithm optimal
- Can we get the node searched in a text output
- Can we get the optimal path
- Is it fast and not too complex
